graphbolt/src/csc_sampling_graph.cc

```diff
@@ -232,11 +265,17 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::SampleNeighbors(
         static_cast<int64_t>(0), std::numeric_limits<int64_t>::max());
     SamplerArgs<SamplerType::LABOR> args{indices_, random_seed, NumNodes()};
     return SampleNeighborsImpl(
```

Comment:
If my template suggestion is used, then the dispatch would happen here with an `if` expression.